iothub/service/src/DirectMethod/DirectMethodsClient.cs

```diff
@@ -0,0 +1,147 @@
+        /// <seealso href="https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-direct-methods"/>
+        public virtual Task<CloudToDeviceMethodResult> InvokeAsync(string deviceId, string moduleId, CloudToDeviceMethod cloudToDeviceMethod, CancellationToken cancellationToken = default)
+        {
+            Argument.RequireNotNullOrEmpty(deviceId, nameof(deviceId));
+            Argument.RequireNotNullOrEmpty(moduleId, nameof(moduleId));
+            Argument.RequireNotNull(cloudToDeviceMethod, nameof(cloudToDeviceMethod));
+
+            return InvokeAsync(GetModuleMethodUri(deviceId, moduleId), cloudToDeviceMethod, cancellationToken);
+        }
+
+        private async Task<CloudToDeviceMethodResult> InvokeAsync(Uri uri,CloudToDeviceMethod cloudToDeviceMethod,CancellationToken cancellationToken)
+        {
+            if (Logging.IsEnabled)
+                Logging.Enter(this, $"Invoking device method for: {uri}", nameof(InvokeAsync));
+
+            try
+            {
+                using HttpRequestMessage request = _httpRequestMessageFactory.CreateRequest(HttpMethod.Post, uri, _credentialProvider, cloudToDeviceMethod);
+                HttpResponseMessage response = await _httpClient.SendAsync(request, cancellationToken);
+                await HttpMessageHelper2.ValidateHttpResponseStatus(HttpStatusCode.OK, response);
+                return await HttpMessageHelper2.DeserializeResponse<CloudToDeviceMethodResult>(response, cancellationToken);
+            }
+            catch (Exception ex)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Error(this, $"{nameof(InvokeAsync)} threw an exception: {ex}", nameof(InvokeAsync));
+                throw;
+            }
+            finally
+            {
+                if (Logging.IsEnabled)
+                    Logging.Exit(this, $"Invoking device method for: {uri}", nameof(InvokeAsync));
+            }
+        }
+
+        private static TimeSpan GetInvokeDeviceMethodOperationTimeout(CloudToDeviceMethod cloudToDeviceMethod)
+        {
+            // For InvokeDeviceMethod, we need to take into account the timeouts specified
+            // for the Device to connect and send a response. We also need to take into account
+            // the transmission time for the request send/receive
+            var timeout = TimeSpan.FromSeconds(15); // For wire time
+            var connectionTimeOut = TimeSpan.FromSeconds(cloudToDeviceMethod.ConnectionTimeoutInSeconds ?? 0);
+            var responseTimeOut = TimeSpan.FromSeconds(cloudToDeviceMethod.ResponseTimeoutInSeconds ?? 0);
+
+            ValidateTimeOut(connectionTimeOut);
+            ValidateTimeOut(responseTimeOut);
+
+            timeout += connectionTimeOut;
+            timeout += responseTimeOut;
+
+            return timeout;
+        }
+
+        private static void ValidateTimeOut(TimeSpan connectionTimeOut)
```

Comment:
The service can do this validation for us, so we don't need to do it on the client side